src/user-support.c

```diff
@@ -123,11 +53,13 @@ void setup_user_xdg_runtime_dir()
 	}
 
 	errno = 0;
-	mkpath(xdg_runtime_dir);
-
-	// if successfully created the directory (ie, not EEXIST), then
-	// chmod it.
+	debug("creating user XDG_RUNTIME_DIR directory: %s", xdg_runtime_dir);
+	if (sc_nonfatal_mkpath(xdg_runtime_dir, 0755) < 0) {
+		die("cannot create user XDG_RUNTIME_DIR directory: %s",
+		    xdg_runtime_dir);
+	}
+	// if successfully created the directory (ie, not EEXIST), then chmod it.
 	if (errno == 0 && chmod(xdg_runtime_dir, 0700) != 0) {
-		die("could not chmod XDG_RUNTIME_DIR");
+		die("cannot change permissions of user XDG_RUNTIME_DIR directory to 0700");
```

Comment:
Sounds fine. This is a little trickier than I made out in this comment so best to leave for another PR.